config/config_params.go

```diff
@@ -338,6 +334,22 @@ func (config *Config) DatastoreConfig() api.CalicoAPIConfig {
 			},
 		}
 	}
+
+	// Build CalicoAPIConfig from the environment.  This means that any XxxYyy field in
+	// CalicoAPIConfigSpec can be set by a corresponding XXX_YYY or CALICO_XXX_YYY environment
+	// variable, and that the datastore type can be set by a DATASTORE_TYPE or
+	// CALICO_DATASTORE_TYPE variable.  (Except in the etcdv2 case which is handled specially
+	// above.)
+	cfg, err := client.LoadClientConfigFromEnvironment()
+	if err != nil {
+		log.WithError(err).Panic("Failed to create datastore config")
```

Comment:
I don't think the situation is actually any different from what it was before; I just simplified the code a bit.  The previous code `client.LoadClientConfig("")` is equivalent to `client.LoadClientConfigFromEnvironment()` (because no file is specified), and I think the latter would be clearer.